src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import cn from 'classnames';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  setFilter: (q: Filter) => void;
+  filterOption: Filter;
+  todos: Todo[];
+}
+
+export const Footer: React.FC<Props> = ({ setFilter, filterOption, todos }) => {
+  const completedTodo = todos.filter(todo => todo.completed).length;
+  const todosLeft = todos.filter(todo => !todo.completed).length;
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {
+          todosLeft === 1
+            ? `${todosLeft} item left`
+            : `${todosLeft} items left`
+        }
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={cn(
+            'filter__link',
+            { selected: filterOption === Filter.All },
+          )}
+          data-cy="FilterLinkAll"
+          onClick={() => setFilter(Filter.All)}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={
+            cn(
+              'filter__link',
+              { selected: filterOption === Filter.Active },
+            )
+          }
+          data-cy="FilterLinkActive"
+          onClick={() => setFilter(Filter.Active)}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={
+            cn(
+              'filter__link',
+              { selected: filterOption === Filter.Active },
+            )
+          }
+          data-cy="FilterLinkCompleted"
+          onClick={() => setFilter(Filter.Completed)}
+        >
+          Completed
+        </a>
+      </nav>
+
+      {/* don't show this button if there are no completed todos */}
+      {completedTodo && (
```

Comment:
`completedTodo` it's not boolean type. In case that you cover `completedTodo` will be `0` and you receive `0` instead nothing:
```
console.log(0 && 'foo');
// 0
```

Options to fix:
1. `!!completedTodo && ...`
2.  `const hasCompleteTodo = todos.some(...);`